### PR TITLE
Broken tests

### PR DESCRIPTION
**Pull request by nikihowe**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

The tests weren't building. Now they are.

## Checklist

- [ ] The documentation is up-to-date with the changes I made.
- [ ] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [ ] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,8 +2,13 @@ CMAKE_MINIMUM_REQUIRED(VERSION 3.3)
 project(polygames)
 
 set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_FLAGS
-    "${CMAKE_CXX_FLAGS} -fsized-deallocation -O3 -ffast-math")
+#set(CMAKE_CXX_FLAGS
```
Comment (nikihowe) [inline CMakeLists.txt]: We will need to remember to set these back once things start working

File: pypolygames/training.py [hunk 2]
```diff
@@ -695,8 +695,11 @@ def run_training(
     optim = create_optimizer(
         model=ddpmodel if ddpmodel is not None else model,
         optim_params=optim_params,
-        optim_state_dict=checkpoint.get("optim_state_dict", None),
+        optim_state_dict=None,  # testing this as per conversation with Adria
```
Comment (nikihowe) [inline pypolygames/training.py]: Do we care about this?

File: pypolygames/training.py [hunk 3]
```diff
@@ -695,8 +695,11 @@ def run_training(
...
     )
+    # print("the ddpmodel was", ddpmodel)
```
Comment (nikihowe) [inline pypolygames/training.py]: These should be deleted probably

File: src/core/game.cc [hunk 4]
```diff
@@ -843,6 +843,7 @@ struct BatchExecutor {
             // state->history().c_str());
             if (randint(256) == 0) {
               fmt::printf(
+                  // "game ended normally, but we're not saying how");
```
Comment (nikihowe) [inline src/core/game.cc]: Delete?

File: src/core/state.h [hunk 5]
```diff
@@ -141,7 +143,7 @@ class State {
 
   template <typename T> void initializeAs() {
     _typeId = &typeid(T);
-    _copyImpl = [](State* dst, const State* src) { *(T*)dst = *(T*)src; };
+    // _copyImpl = [](State* dst, const State* src) { *(T*)dst = *(T*)src; };
```
Comment (nikihowe) [inline src/core/state.h]: Is this a permanent solution?

File: src/core/state.h [hunk 6]
```diff
@@ -632,7 +634,11 @@ class State {
   bool _stochasticReset;
 
   const std::type_info* _typeId = nullptr;
-  void (*_copyImpl)(State* dst, const State* src) = nullptr;
+  // void (*_copyImpl)(State* dst, const State* src) = nullptr;
```
Comment (nikihowe) [inline src/core/state.h]: See above question

File: src/games/nogo_zestate.h [hunk 7]
```diff
@@ -77,13 +77,16 @@ class StateForNogo : public State {
...
     assert(_status != GameStatus::player0Win);
     assert(_status != GameStatus::player1Win);
-    NoGoAction nogoAction(_nogoGame.GetTurnPlayer(), action.GetHash());
+    // NOTE: nhowe changed three occurrences of "GetHash" to "GetIndex",
+    //       which seems more correct. This was the only place that
+    //       "GetHash" was used.
+    NoGoAction nogoAction(_nogoGame.GetTurnPlayer(), action.GetIndex());
     if (_nogoGame.GetTurnPlayer() == PLAYER_0) {
-      _features[action.GetHash()] = 1.;
+      _features[action.GetIndex()] = 1.;
```
Comment (nikihowe) [inline src/games/nogo_zestate.h:85]: Would be nice to make sure this didn't break anything

File: tests/havannah-state-tests.cc [hunk 8]
```diff
@@ -21,8 +22,16 @@ namespace Havannah {
...
-    Havannah::State<SIZE, PIE, EXTENDED>(seed, history, turnFeatures) {}
-   GameStatus GetStatus() { return ::State::_status; };
+   Havannah::State<SIZE, PIE, EXTENDED>(seed) {
+      // nhowe added
+      // currently, actively, trying to get the turnFeatures to work properly given it's deprecated
+      // std::list<core::FeatureOptions> featopts;
+      // featopts.emplace_back();
+      // core::FeatureOptions& opt = featopts.back();
+      // opt.turnFeaturesMultiChannel = turnFeatures;
+      // state_->setFeatures(&opt);
```
Comment (nikihowe) [inline tests/havannah-state-tests.cc:32]: Come back to this later once we've got one game working

**Review by nikihowe — COMMENTED** [review]
Would be good to get your thoughts on these comments @rhaps0dy

File: CMakeLists.txt (hunk 1 above)
Comment (rhaps0dy) [reply CMakeLists.txt]: Sure I'll add conditional ones.

We don't need -fsized-deallocation https://developer.arm.com/documentation/101754/0619/armclang-Reference/armclang-Command-line-Options/-fsized-deallocation---fno-sized-deallocation

File: pypolygames/training.py (hunk 2 above)
Comment (rhaps0dy) [reply pypolygames/training.py]: maybe not IDK? Let's undo it.

File: src/core/state.h (hunk 5 above)
Comment (rhaps0dy) [reply src/core/state.h]: Now that you mention it is not

File: src/games/nogo_zestate.h (hunk 7 above)
Comment (rhaps0dy) [reply src/games/nogo_zestate.h:85]: Yeah IDK maybe this is what the state tests are for?

**Review by rhaps0dy — APPROVED** [review]
It was a huge pain but phew

**Pull request merged**